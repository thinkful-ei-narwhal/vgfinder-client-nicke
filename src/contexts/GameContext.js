import React, { Component } from "react";

export const nullGame = {
  id: "",
  title: "",
  description: "",
  genre: "",
  rating: "",
  release_date: "",
  developer: "",
  trailer_url: "",
  image_url_box_art: "",
  image_url_two: "",
  image_url_three: "",
  image_url_four: "",
  image_url_five: "",
};

const GameContext = React.createContext({
  game: nullGame,
  setGame: () => {},
  clearGame: () => {},
});

export default GameContext;

export class GameProvider extends Component {
  state = {
    game: nullGame,
  };

  setGame = (game) => {
    this.setState({ game });
  };

  clearGame = () => {
    this.setGame(nullGame);
  };

  render() {
    const value = {
      game: this.state.game,
      setGame: this.setGame,
      clearGame: this.clearGame,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}
