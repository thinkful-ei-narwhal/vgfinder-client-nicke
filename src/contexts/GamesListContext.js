import React, { Component } from 'react';

const GamesListContext = React.createContext({
  gamesList: [],
  error: null,
  setError: () => { },
  clearError: () => { },
  setGamesList: () => { },
})
export default GamesListContext

export class GamesListProvider extends Component {
  state = {
    gamesList: [],
    error: null,
  };

  setGamesList = gamesList => {
    this.setState({ gamesList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      gamesList: this.state.gamesList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGamesList: this.setGamesList,
    }
    return (
      <GamesListContext.Provider value={value}>
        {this.props.children}
      </GamesListContext.Provider>
    )
  }
}
