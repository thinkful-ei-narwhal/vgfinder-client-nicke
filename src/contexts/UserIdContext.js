import React, { Component } from 'react'

const UserIdContext = React.createContext({
  userId: undefined,
  wishlist: [],
  setUserId: () => { },
  setUserWishlist: () => { },
})

export default UserIdContext

export class UserIdContextProvider extends Component {
  state = {
    userId: undefined,
    wishlist: []
  };

  setUserId = (userId) => {
    this.setState({ userId })
  }

  setUserWishlist = (wishlist) => {
    this.setState({ wishlist })
  }


  render() {
    const value = {
      userId: this.state.userId,
      wishlist: this.state.wishlist,
      setUserId: this.setUserId,
      setUserWishlist: this.setUserWishlist
    }

    return (
      <UserIdContext.Provider value={value}>
        {this.props.children}
      </UserIdContext.Provider>
    )
  }
}
