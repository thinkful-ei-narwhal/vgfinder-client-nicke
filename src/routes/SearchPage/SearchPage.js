import './SearchPage.css';
import React, { Component } from 'react';
import Header from '../../components/Header/Header';
import { Input, Section } from './../../components/Utils/Utils';
import GamesApiService from '../../services/games-api-service';
import GamesListContext from '../../contexts/GamesListContext';
import Search from '../../components/Search/Search'

export default class SearchPage extends Component {
  static contextType = GamesListContext;

  constructor(props) {
    super(props)
    this.state = {
      loaded: false,
    }
  }

  componentDidMount() {
    this.context.clearError()
    GamesApiService.getGames()
      .then(res => {
        this.context.setGamesList(res)
        this.setState({ loaded: true, games: res })
      })
      .catch(this.context.setError)
  }

  renderSearchPage() {
    return (
      <>
        <Header pathName={this.props.location.pathname} />
        <Search gamesList={this.context.gamesList} />
      </>
    );
  }

  render() {
    return (
      <>

        {this.state.loaded ? this.renderSearchPage() : null}
      </>
    )
  }

}