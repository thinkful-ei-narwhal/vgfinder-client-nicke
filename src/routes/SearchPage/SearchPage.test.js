import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchPage from './SearchPage'
import { GamesListProvider } from './../../contexts/GamesListContext'

describe(`SearchPage component`, () => {
  it('renders a SearchPage component', () => {
    const wrapper = shallow(
      <GamesListProvider>
        <SearchPage />
      </GamesListProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
