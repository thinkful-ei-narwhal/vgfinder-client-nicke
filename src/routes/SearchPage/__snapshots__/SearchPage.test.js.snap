// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage component renders a SearchPage component 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "gamesList": Array [],
      "setError": [Function],
      "setGamesList": [Function],
    }
  }
>
  <SearchPage />
</ContextProvider>
`;
